src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,130 @@
-import React from 'react';
+import React, {
+  useState, useEffect, useCallback,
+} from 'react';
+import { debounce } from 'lodash';
+import classNames from 'classnames';
+import { TodoListProps } from '../../types';
 import './TodoList.scss';
+import { getTodos } from '../../api';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
+export const TodoList: React.FC<TodoListProps> = ({ todos, onUserSelect }) => {
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [filteredTodos, setFilteredTodos] = useState(todos);
```

Comment:
why do you need save 2 todo arrays in state?
just prepare filtered before rendering them